Source/MQTTnet/Server/MqttServer.cs

```diff
@@ -121,7 +121,8 @@ public async Task StartAsync(IMqttServerOptions options)
 
             _cancellationTokenSource = new CancellationTokenSource();
 
-            _retainedMessagesManager = new MqttRetainedMessagesManager(Options, _logger);
+            _retainedMessagesManager = Options.RetainedMessagesManager ?? new MqttRetainedMessagesManager();
```

Comment:
I suggest to move that into the options class only. So that the property is initialized with the default manager. If the value is null here we can throw a proper exception.

Please consider adding a new exception like "MqttConfigurationException : MqttException" which can be used for other invalid config scenarios as well. I would like to use such an exception for other places like when MQTTv5 features are used but V3.1.1 is selected in client options.

Reply:
Done in new patch. (You probably want to check the exception message.)

Reply:
Also, MqttException does not exists. I use Exception as the other exceptions in MQTTnet do.